include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -817,14 +817,14 @@ class __future : private std::tuple<_Args...>
             auto& __val = std::get<0>(*this);
             return __wait_and_get_value(__val);
         }
-        else
-            wait();
+
+        wait();
```

Comment:
Is there some value in this `else`, since it is attached to a `constexpr` conditional? 

Perhaps its eliminated as dead code either way, but I tend to prefer to keep these `else` blocks anyway when we are not battling very large number of indents anyway for clarity, but here we can help the compiler rather than relying on dead code elimination I think.

Reply:
this change has been removed.